crates/json/src/schema/formats.rs

```diff
@@ -22,73 +60,161 @@ lazy_static::lazy_static! {
+            Self::RelativeJsonPointer => {
+                ValidationResult::from(RELATIVE_JSON_POINTER_RE.is_match(val).unwrap_or(false))
             }
         }
-        "ipv6" => ValidationResult::from(match IpAddr::from_str(val) {
-            Ok(i) => ValidationResult::from(i.is_ipv6()),
-            Err(e) => ValidationResult::Invalid(Some(e.to_string())),
-        }),
-        // uuid crate supports non-hyphenated inputs, jsonschema does not
-        "uuid" if val.len() == 36 => ValidationResult::from(Uuid::parse_str(val)),
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::Format;
+    use crate::validator::ValidationResult;
+
+    #[test]
+    fn test_format_cases() {
+        // Missing format cases:
+        //  * idn-hostname
+        //  * idn-email
+        //  * iri
+        //  * iri-reference
+        for (format, value, expect) in [
+            ("date", "2022-09-11", true),
+            ("date", "2022-09-11T10:31:25.123Z", false),
+            ("date-time", "2022-09-11T10:31:25.123Z", true),
+            ("date-time", "10:31:25.123Z", false),
+            ("time", "10:31:25.123Z", true),
+            ("email", "john@doe.com", true),
+            ("email", "john at doe.com", false),
+            ("hostname", "hostname.com", true),
+            ("hostname", "hostname dot com", false),
+            ("ipv4", "123.45.6.78", true),
+            ("ipv4", "123.45.6.78.9", false),
+            ("ipv4", "0.1.2.3", false),
+            ("ipv6", "2001:0db8:0000:0000:0000:ff00:0042:8329", true),
+            ("ipv6", "2001:db8::ff00:42:8329", true),
+            ("ipv6", "2001 db8  ff00:42:8329", false),
+            ("macaddr", "001B638445E6", true),
+            ("macaddr", "00:1b-63:84-45:e6", true),
+            ("macaddr", "00!1b!63:84!45:e6", false),
+            ("macaddr", "00:1b:63:84:45:e6", true),
+            ("macaddr8", "00:1b:63:84:45:e6", false),
+            ("macaddr8", "00:1b:63:84:45:e6:aa:bb", true),
+            ("macaddr8", "00-1b-638445e6aa:bb", true),
+            ("uuid", "df518555-34f0-446a-8788-7b36f607bbea", true),
+            ("uuid", "DF518555-34F0-446A-8788-7B36F607BBEA", true),
+            ("uuid", "not-a-UUID-7B36F607BBEA", false),
+            ("duration", "P1M3DT30H4S", true),
+            ("duration", "P1W", true),
+            ("duration", "P1W3D", false), // Mixes weeks and days (disallowed).
+            ("duration", "roundtuit", false),
+            ("uri", "http://www.example.org/foo/bar", true),
+            ("uri", "../path/to/bar", false),
+            ("uri-reference", "../path/to/bar", true),
+            ("uri", "http://example.com/~{username}/", false),
+            ("uri-template", "http://example.com/~{username}/", true),
+            ("json-pointer", "/valid/json pointer", true),
+            ("json-pointer", "/invalid/es~cape", false),
+            ("relative-json-pointer", "0/objects", true),
+            ("regex", "^hello$", true),
+            ("regex", "[hello", false),
```

Comment:
These are great, though FWIW the formats are more exhaustively tested by the optional json-schema format tests that I pulled in [here](https://github.com/estuary/flow/blob/master/crates/json/tests/draft2019_format_tests.rs)

Reply:
They totally are, this is just a lightweight supplement / sanity check to that, and a place to cover `macaddr` / `macaddr8` that naturally isn't covered there.